### ISSUE TITLE
Retrieval of test status on "Status" screen in low bandwidth situations

### ISSUE THREAD

**Issue by jkrwdf** [corona-warn-app/cwa-app-android#4847]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [?] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Google Pixel 3 XL
- Android version: 12
- App version: 2.17.2

## Describe the bug

The result of a test (for example rapid-antigen-test) which is registered in the CWA is eventually not shown on the "Status" page when the phone is in a low bandwidth network (e.g. EDGE). 

When you start CWA in this situation, you see "Datenabruf... Das Ergebnis wird aktualisiert" for potentially long time.

![image](https://user-images.githubusercontent.com/5329491/154437269-05ad83d4-ed6b-4f8b-bad5-bf70c115cc9b.png)

I classify this behavior as bug because this behavior is also present when the result of the test _is_ already known to the CWA. 
Proof: When I switch off all networks and start CWA then, the result is shown _directly_. 

In situations where there is only EDGE available and when you need to present your test result using CWA, this may be an issue. 

## Steps to reproduce the issue

Register test (e.g. RAT)
Confirm that test result is fetched by CWA. 
Displace CWA from memory.
Switch to EDGE or GPRS
Start CWA

## Expected behaviour

When the result of the test has already been retrieved by the CWA, it shall be shown immideately. 

## Possible Fix

The repetitive attempt to fetch the result fresh from the server is IMHO redundant. 

If this suggestion is not possible for any reason, my alternative suggestion is to have a short timeout (max. 5 seconds) before you display the cached test result. Currently, the timeout is in the order of a minute.

---
Internal Tracking ID: [EXPOSUREAPP-11895](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11895)

**Comment by Ein-Tim** [comment]
Hey @jkrwdf

Did you take a look at #4338? To me it seems like this is the same issue, or?

**Comment by jkrwdf** [comment]
Issue #4338 refers to the situation that there is _no_ internet connection. 

In this situation (e.g. when I go to flight-mode), I observe that the test result is shown directly (and can be tapped so I come to the screen with the big clock and my name and birthday). 

My situation describes a network connection which is very slow, so CWA does not fall back to this "No Internet" mode, but tries for a felt minute to refresh the state, although it already knows it.

**Comment by dsarkar** [comment]
@jkrwdf Thanks for the report. Just for your information, and without assessing at this stage if your report  is to be considered a bug or not: The result is retrieved not only once, since it is possible to change the test result afterwards in certain situations.

**Comment by dsarkar** [comment]
Internal Tracking ID: [EXPOSUREAPP-11895](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11895)

Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by vaubaehn** [comment]
@dsarkar @jkrwdf @Ein-Tim 
Just a short feedback: it looks like the problem described here is more close to #4148 which was enhanced/fixed by @mtwalli 's #4151 . The problem is a network call that gets stuck on a slow connection. The solution for the issue reported by @jkrwdf could be a similar `getCurrentTestResukt.first()` way.
For a long-term app improvement it could be good to assess all network calls whether they can be enhanced in a similar way.

**Comment by dsarkar** [comment]
@vaubaehn Thanks for your analysis and pointing related issues/PR out. We will forward this.

**Comment by dsarkar** [comment]
So, I understand app's behaviour is currently as specified.

**Comment by Ein-Tim** [comment]
@dsarkar Have the devs also already assessed the situation in https://github.com/corona-warn-app/cwa-app-android/issues/4338, which seems related to this issue?

**Comment by dsarkar** [comment]
@Ein-Tim  Regarding https://github.com/corona-warn-app/cwa-app-android/issues/4338, yes, there are some internal discussions, I will come back to you.
